tests/tls/run.sh

```diff
@@ -47,9 +47,114 @@ function prepare_data() {
+	# only mysql 8.0 support use `ALTER INSTANCE RELOAD TLS` to reload cert
+	# when use mysql 5.7 we need to restart mysql-server manually if your local server do not enable ssl
+
+	run_sql "grant all on *.* to 'dm_tls_test'@'%' identified by '123456' require ssl;" $MYSQL_PORT1 $MYSQL_PASSWORD1
+	run_sql 'flush privileges;' $MYSQL_PORT1 $MYSQL_PASSWORD1
+
+	cp $cur/conf/source1.yaml $WORK_DIR/source1.yaml
+	sed -i "s%dir-placeholer%$mysql_data_path%g" $WORK_DIR/source1.yaml
+	echo "add dm_tls_test user done $mysql_data_path"
+}
+
+function test_worker_ha_when_enable_source_tls() {
+	cleanup_data tls
+	cleanup_process
+
+	setup_mysql_tls
 	run_tidb_with_tls
+	prepare_data
+
+	cp $cur/conf/dm-master1.toml $WORK_DIR/
+	cp $cur/conf/dm-master2.toml $WORK_DIR/
+	cp $cur/conf/dm-master3.toml $WORK_DIR/
+	cp $cur/conf/dm-worker1.toml $WORK_DIR/
+	cp $cur/conf/dm-worker2.toml $WORK_DIR/
+	cp $cur/conf/dm-task.yaml $WORK_DIR/
+
+	sed -i "s%dir-placeholer%$cur\/conf%g" $WORK_DIR/dm-master1.toml
+	sed -i "s%dir-placeholer%$cur\/conf%g" $WORK_DIR/dm-master2.toml
+	sed -i "s%dir-placeholer%$cur\/conf%g" $WORK_DIR/dm-master3.toml
+	sed -i "s%dir-placeholer%$cur\/conf%g" $WORK_DIR/dm-worker1.toml
+	sed -i "s%dir-placeholer%$cur\/conf%g" $WORK_DIR/dm-worker2.toml
+	sed -i "s%dir-placeholer%$cur\/conf%g" $WORK_DIR/dm-task.yaml
+
+	run_dm_master $WORK_DIR/master1 $MASTER_PORT1 $WORK_DIR/dm-master1.toml
+	run_dm_master $WORK_DIR/master2 $MASTER_PORT2 $WORK_DIR/dm-master2.toml
+	run_dm_master $WORK_DIR/master3 $MASTER_PORT3 $WORK_DIR/dm-master3.toml
+	check_rpc_alive $cur/../bin/check_master_online 127.0.0.1:$MASTER_PORT1 "$cur/conf/ca.pem" "$cur/conf/dm.pem" "$cur/conf/dm.key"
+	check_rpc_alive $cur/../bin/check_master_online 127.0.0.1:$MASTER_PORT2 "$cur/conf/ca.pem" "$cur/conf/dm.pem" "$cur/conf/dm.key"
+	check_rpc_alive $cur/../bin/check_master_online 127.0.0.1:$MASTER_PORT3 "$cur/conf/ca.pem" "$cur/conf/dm.pem" "$cur/conf/dm.key"
+
+	run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $WORK_DIR/dm-worker1.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT "$cur/conf/ca.pem" "$cur/conf/dm.pem" "$cur/conf/dm.key"
+
+	# operate mysql config to worker
+	run_dm_ctl_with_tls_and_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" $cur/conf/ca.pem $cur/conf/dm.pem $cur/conf/dm.key \
+		"operate-source create $WORK_DIR/source1.yaml" \
+		"\"result\": true" 2 \
+		"\"source\": \"$SOURCE_ID1\"" 1
 
+	#  start task
+	echo "start task and check stage"
+	run_dm_ctl_with_tls_and_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" $cur/conf/ca.pem $cur/conf/dm.pem $cur/conf/dm.key \
+		"start-task $WORK_DIR/dm-task.yaml" \
+		"\"result\": true" 2
+
+	run_dm_ctl_with_tls_and_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" $cur/conf/ca.pem $cur/conf/dm.pem $cur/conf/dm.key \
+		"query-status test" \
+		"\"result\": true" 2
+
+	echo "check data"
+	check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+
+	echo "pause task before kill and restart dm-worker"
```

Comment:
I think we should delete tls files first to make sure we have ha.

Reply:
I don't think so, the worker doesn't read the certificate from the file in the code now, 
we always read it from etcd, because we use this method to load the certificate (ToTLSConfigWithVerifyByRawbytes)

Reply:
If dm-worker reads tls files from the local disk, we can also pass this test. Our test should make sure the former way fails this test.

Reply:
also added in https://github.com/pingcap/dm/pull/1910/commits/6d62706696b724e168272dd7e7edcae00828cdab